### DISCUSSION TITLE
change output resolution in stream io

### DISCUSSION THREAD

**Discussion by sugizo** [python-ffmpegio/python-ffmpegio#38] [Q&A] [answered]
**expected result**
small video output resolution 

**code**
```
input = 'http://195.196.36.242/mjpg/video.mjpg'
output = 'rtsp.mp4'
i = 0
with ffmpegio.open(input, 'rv') as fin, ffmpegio.open(output, 'wv', rate_in = fin.rate, shape_in = (180, 320), overwrite = True) as fout:
    for frames in fin:
        fout.write(frames)
        if i == 9:
            break
        i += 1
```

**result** 
no error occured but the result is not expected, 
video resolution is still high

**expected result**
video output resolution is small as mention in args 180 x 320

**p.s.**
tryin to use shape args on both open('rv') and open'('wv') but return an error
tryin to use shape_in args on open('rv') but return an error
tryin to use (1, 360, 640, 3) as value but result is not expected (video resolution is still high

best regards

**Comment by tikuma-lsuhsc** [comment, marked as answer]
The correct approach is to make the reader to output small frames:

```python
with ffmpegio.open(input, 'rv', s=(180,320)) as fin, ffmpegio.open(output, 'wv', rate_in = fin.rate, overwrite = True) as fout:
    ...
```

The FFmpeg `s` option (or `-s` option in cli) sets the output frame size (implicitly inserts a resize filter). If you wish to only set height or width and automatically scale the other, keeping the aspect ratio, specify the dependent dimension as `-1` or `-2` if you need to make sure even number of pixels. For example,

```python
with ffmpegio.open(input, 'rv', s=(180,-1)) as fin, ffmpegio.open(output, 'wv', rate_in = fin.rate, overwrite = True) as fout:
    ...
```
